connector/src/main/java/org/geysermc/connector/network/translators/java/entity/player/JavaPlayerPositionRotationTranslator.java

```diff
@@ -95,20 +96,59 @@ public void translate(ServerPlayerPositionRotationPacket packet, GeyserSession s
 
         session.setSpawned(true);
 
-        if (!packet.getRelative().isEmpty()) {
-            session.setTeleportCache(new TeleportCache(packet.getX(), packet.getY(), packet.getZ(), packet.getTeleportId()));
-            entity.moveRelative(session, packet.getX(), packet.getY() + EntityType.PLAYER.getOffset() + 0.1f, packet.getZ(), packet.getYaw(), packet.getPitch(), true);
+        double newX;
+        double newY;
+        double newZ;
+
+        double newPitch;
+        double newYaw;
+
+        // If coordinates are relative, then add to the existing coordinate
+        if (packet.getRelative().contains(PositionElement.X)) {
```

Comment:
I can't see a way where we don't need all these if statements, but it would probably look cleaner if you did:

`double newX = packet.getX + (packet.getRelative().contains(PositionElement.X) ? entity.getPosition.getX() : 0);`

Reply:
I'll do that